clean/baker_pd_cprr.py

```diff
@@ -62,24 +69,133 @@ def clean_allegation(df):
 
 
-def clean():
+def clean_dispositions17(df):
+    df.loc[:, "not_sustained"] = (
+        df.not_sustained.astype(str)
+        .str.lower()
+        .str.strip()
+        .fillna("")
+        .str.replace(r"^yes$", "not sustained", regex=True)
+        .str.replace(r"^nan$", "", regex=True)
+    )
+
+    df.loc[:, "unknown"] = (
+        df.unknown.astype(str)
+        .str.lower()
+        .str.strip()
+        .fillna("")
+        .str.replace(r"^yes$", "unknown", regex=True)
+        .str.replace(r"^nan$", "", regex=True)
+    )
+
+    df.loc[:, "sustained"] = (
+        df.sustained.astype(str)
+        .str.lower()
+        .str.strip()
+        .fillna("")
+        .str.replace(r"^yes$", "sustained", regex=True)
+        .str.replace(r"^nan$", "", regex=True)
+    )
+
+    unknowns = df.unknown.str.extract(r"(unknown)")
+    not_sustained = df.not_sustained.str.extract(r"(not sustained)")
+    sustained = df.sustained.str.extract(r"(sustained)")
+
+    df.loc[:, "disposition_1"] = unknowns[0]
+    df.loc[:, "disposition_2"] = not_sustained[0]
+    df.loc[:, "disposition_3"] = sustained[0]
+
+    df.loc[:, "disposition"] = df.disposition_1.fillna("").str.cat(
+        df.disposition_2.fillna(""), sep=""
+    )
+
+    df.loc[:, "disposition"] = df.disposition.str.cat(
+        df.disposition_3.fillna(""), sep=""
+    )
+
+    return df.drop(
+        columns=[
+            "disposition_1",
+            "disposition_2",
+            "disposition_3",
+            "unknown",
+            "sustained",
+            "not_sustained",
+        ]
+    )
+
+
+def split_rows_with_multiple_allegations(df):
+    df = (
+        df.drop("allegation", axis=1)
+        .join(
+            df["allegation"]
+            .str.split(";", expand=True)
+            .stack()
+            .reset_index(level=1, drop=True)
+            .rename("allegation"),
+            how="outer",
+        )
+        .reset_index(drop=True)
+    )
+    return df
+
+
+def clean20():
     df = (
         pd.read_csv(deba.data("raw/baker_pd/baker_pd_cprr_2018_2020.csv"))
         .pipe(clean_column_names)
         .pipe(clean_allegation)
-        .pipe(join_disposition_columns)
+        .pipe(join_disposition_columns20)
         .pipe(split_names)
         .pipe(set_values, {"agency": "Baker PD"})
         .pipe(gen_uid, ["first_name", "last_name", "agency"])
         .pipe(gen_uid, ["uid", "disposition", "allegation"], "allegation_uid")
+        .drop_duplicates(subset=["allegation_uid"])
+
+    )
+    return df
+
+
+def clean17():
+    df = (
+        pd.read_csv(
+            deba.data("raw/baker_pd/baker_pd_cprr_2014_2017.csv"), encoding="cp1252"
+        )
+        .pipe(clean_column_names)
+        .pipe(split_names)
+        .pipe(clean_allegation)
+        .pipe(split_rows_with_multiple_allegations)
+        .pipe(clean_dispositions17)
+        .pipe(standardize_desc_cols, ["allegation"])
+        .pipe(set_values, {"agency": "Brown PD"})
```

Comment:
Is it really "Brown PD"?

Reply:
This is ready for review as well. Thanks!